ToDoList-Redux/Binders/TaskListViewModel.swift

```diff
@@ -65,25 +57,25 @@ extension Reducer where ActionType == TaskList.Action, StateType == TaskList.Sta
 // MARK: - PROJECTIONS
 
 extension CheckmarkCellView {
-    
     static func viewModel<S: StoreType>(store: S, taskId: String) -> ObservableViewModel<CheckmarkCellView.Action, CheckmarkCellView.State>
-    where S.ActionType == TaskList.Action, S.StateType == TaskList.State {
-        let task = store.mapState { state in state.tasks.first { $0.id == taskId } }
+    where S.ActionType == TaskAction, S.StateType == [Task] {
+        let task = store.mapState { state in state.first { $0.id == taskId } }
         return task
-            .projection(action: Self.transform, state: Self.transform)
+            .projection(action: Self.transform(taskId: taskId), state: Self.transform)
```

Comment:
Here is interesting, we already have the task id in this context, we just supply our transform with it.